test/denormalized_has_many_test.rb

```diff
@@ -141,6 +141,40 @@ def test_deprecated_set_inverse_association_on_cache_hit
     assert_equal item.object_id, associated_record.item.object_id
   end
 
+  def test_returned_records_should_be_readonly_on_cache_hit
+    IdentityCache.with_fetch_read_only_records do
+      Item.fetch(@record.id) # warm cache
+      record_from_cache_hit = Item.fetch(@record.id)
+      assert record_from_cache_hit.fetch_associated_records.all?(&:readonly?)
+    end
+  end
+
+  def test_returned_records_should_be_readonly_on_cache_miss
+    IdentityCache.with_fetch_read_only_records do
+      record_from_cache_miss = Item.fetch(@record.id)
+      assert record_from_cache_miss.fetch_associated_records.all?(&:readonly?)
+    end
+  end
+
+  def test_db_returned_records_should_never_be_readonly
+    IdentityCache.with_fetch_read_only_records do
+      record_from_db = Item.find(@record.id)
+      uncached_records = record_from_db.associated_records
+      refute uncached_records.all?(&:readonly?)
+      record_from_db.fetch_associated_records
+      refute uncached_records.all?(&:readonly?)
```

Comment:
Oh, I see.  I had a different idea of what would be marked read-only.  Since `record_from_db.fetch_associated_records` wouldn't ever fetch records from the cache here, I figured the records don't need to be marked read-only.  But since you aren't marking the active record association's records as read-only, then this should be :ok: